generators/generator-microsoft-bot-adaptive/generators/app/index.js

```diff
@@ -228,64 +152,101 @@ module.exports = class extends Generator {
     );
   }
 
-  _copySchemas() {
-    const botName = this.options.botName;
-    const directoryName = 'schemas';
+  _writeJsPackageJson() {
+    const { botName, integration } = this.options;
 
-    this.fs.copyTpl(
-      this.templatePath(path.join('assets', directoryName)),
-      this.destinationPath(path.join(botName, directoryName)),
-      {
-        botName,
-      }
-    );
+    const dependencies = {};
+
+    switch (integration) {
+      case integrations.webapp:
+        Object.assign(dependencies, {
+          'botbuilder-runtime-integration-restify': 'next',
+          minimist: 'latest',
+        });
+
+        break;
+
+      case integrations.functions:
+        Object.assign(dependencies, {
+          'botbuilder-runtime-integration-azure-functions': 'next',
+        });
+
+        break;
+    }
+
+    this.fs.writeJSON(this.destinationPath(botName, 'package.json'), {
+      name: botName,
+      private: true,
+      scripts: {
+        build: 'echo done',
+      },
+      dependencies: Object.assign(
+        {},
+        this.packageReferences.reduce(
+          (acc, { name, version }) => Object.assign(acc, { [name]: version }),
+          {}
+        ),
+        dependencies
+      ),
+    });
   }
 
   _writeApplicationSettings() {
-    const botName = this.options.botName;
-    const fileName = 'appsettings.json';
-
-    const filePath = path.join(
+    const {
+      applicationSettingsDirectory = '.',
       botName,
-      this.applicationSettingsDirectory
-        ? this.applicationSettingsDirectory
-        : '',
-      fileName
-    );
+      integration,
+      platform,
+    } = this.options;
 
     const appSettings = this.fs.readJSON(
-      this.templatePath(path.join('assets', fileName))
+      this.templatePath('assets', 'appsettings.json')
     );
 
     appSettings.luis.name = botName;
-    appSettings.runtime.command = `dotnet run --project ${botName}.csproj`;
-    appSettings.runtime.key = 'adaptive-runtime-dotnet-webapp';
 
-    for (const pluginDefinition of this.pluginDefinitions) {
-      appSettings.runtimeSettings.plugins.push(pluginDefinition);
+    appSettings.runtime.key = `adaptive-runtime-${platform}-${integration}`;
+    appSettings.runtime.command = {
+      [platforms.dotnet]: `dotnet run --project ${botName}.csproj`,
+      [platforms.js]: 'node index.js',
+    }[platform];
+
+    for (const { name } of this.packageReferences) {
+      appSettings.runtimeSettings.plugins.push({
+        name,
+        settingsPrefix: name,
+      });
     }
 
-    if (this.modifyApplicationSettings) {
-      this.modifyApplicationSettings(appSettings);
+    for (const { name, settingsPrefix } of this.pluginDefinitions) {
+      appSettings.runtimeSettings.plugins.push({
+        name,
+        settingsPrefix: settingsPrefix || name,
+      });
     }
```

Comment:
Do we need to add packageReferences? Won't this result in duplication given the current state of the child generators?

Reply:
It will indeed - `packageReferences` do need to be included in the `plugins` array due to how the runtime (at least in JS, and I believe dotnet as well) discovers plugins. We _only_ [iterate over the `runtimeSettings.plugins` array](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-runtime/src/index.ts#L198). If packages aren't referenced there they will not load.

Reply:
I will remove dupes from child generators though as we don't want dupes in the plugins list.

Reply:
@peterinnesmsft due to the 100% overlap in package references and plugin definitions, I decided to consolidate the two settings into a single `packageReferences` setting with `name`, `version`, and optional `settingsPrefix`. This made sense to me, let me know if you disagree.